src/fprime/fbuild/interaction.py

```diff
@@ -69,41 +74,100 @@ def run_impl(deployment: Path, path: Path, platform: str, verbose: bool):
+    """ Adds new component or port to CMakeLists.txt"""
+    print("[INFO] Found CMakeLists.txt at '{}'".format(list_file))
+    with open(list_file, "r") as f:
+        lines = f.readlines()
+
+    addition = (
+        'add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/' + str(comp_path) + '/")\n'
+    )
+    if addition in lines:
+        print("Already added to CMakeLists.txt")
+        return True
+
+    if not confirm(
+        "Add component {} to {} {}?".format(comp_path, list_file, "at end of file?")
+    ):
+        return False
+
+    lines.append(addition)
+    with open(list_file, "w") as f:
+        f.write("".join(lines))
+    return True
+
+
+@contextmanager
+def suppress_stdout():
+    with open(os.devnull, "w") as devnull:
+        old_stdout = sys.stdout
+        sys.stdout = devnull
+        try:
+            yield
+        finally:
+            sys.stdout = old_stdout
+
+
+def regenerate(build: Build):
+    handler = CMakeHandler()
+    print("Refreshing cache to include new addition...")
+    with suppress_stdout():
+        handler.cmake_refresh_cache(build.get_build_cache())
+
+
+def add_unit_tests(deployment, comp_path, platform, verbose):
+    # Creates unit tests and moves them into test/ut directory
+    os.chdir(str(comp_path))
+    if confirm("Would you like to generate unit tests?: "):
+        test_path = Path("test", "ut")
+        test_path.mkdir(parents=True, exist_ok=True)
+        target = Target.get_target("impl", {"ut"})
+        build = Build(target.build_type, deployment, verbose=verbose)
+        build.load(comp_path, platform)
+        print("Generating unit tests...")
+        with suppress_stdout():
+            build.execute(target, context=comp_path, make_args={})
+        test_files = [
+            "Tester.hpp",
+            "Tester.cpp",
+            "TesterBase.hpp",
+            "TesterBase.cpp",
+            "GTestBase.hpp",
+            "GTestBase.cpp",
+            "TestMain.cpp",
+        ]
+        for file in test_files:
+            if os.path.isfile(file):
+                new_name = test_path / file
+                os.rename(file, str(new_name))
+
+
+def add_port_to_cmake(list_file: Path, comp_path: Path):
+    """ Adds new port to CMakeLists.txt in port directory"""
     print("[INFO] Found CMakeLists.txt at '{}'".format(list_file))
     with open(list_file, "r") as file_handle:
         lines = file_handle.readlines()
-    topology_lines = [
-        (line, text) for line, text in enumerate(lines) if "/Top/" in text
-    ]
-    line = len(topology_lines)
-    if topology_lines:
-        line, text = topology_lines[0]
-        print(
-            "[INFO] Topology inclusion '{}' found on line {}.".format(
-                text.strip(), line + 1
-            )
-        )
+    index = 0
+    while re.search("set\(\s*SOURCE_FILES", lines[index]) is None:
+        index += 1
+    index += 1
+    while re.search("CMAKE_CURRENT_LIST_DIR", lines[index]):
```

Comment:
If you're not using regex patterns, it might be simpler to go with a substring search instead:

```
if "CMAKE_CURRENT_LIST_DIR" in lines[index]:
```